### PR TITLE
Update babel and eslint-config-airbnb

### LINKED ISSUE #204: Update to babel 6
Tried once when it just was out and it broke in many ways. Lets try again, now that things settled.

We'd also update other dependencies like eslint and co at the same time. This should not need any code changes.

### PR DESCRIPTION
**Pull request by marco-c**
Fixes #204.

### CODE REVIEW

File: engine/browserParser.js [hunk 1]
```diff
@@ -1,18 +1,20 @@
 import cache from './cache.js';
 
 export default class BrowserParser {
-  results = {
```
Comment (digitarald) [inline engine/browserParser.js]: Shouldn't this work when adding `stage-0` (`stage-1` has `class-properties`)

Comment (marco-c) [inline engine/browserParser.js]: Wouldn't this only work if we added stage-1?

The linter fails without this change:
```
[15:39:18] Starting 'lint'...
[15:39:18] 'lint' errored after 337 ms
[15:39:18] ESLintError in plugin 'gulp-eslint'
Message:
    Parsing error: Unexpected token =
Details:
    fileName: engine/browserParser.js
    lineNumber: 4
[15:39:18] 
engine/browserParser.js
  4:12  error  Parsing error: Unexpected token =

✖ 1 problem (1 error, 0 warnings)
```

Comment (digitarald) [inline engine/browserParser.js]: stage-0 includes stage-1. seems like the linter doesn't know about stage-1, or specifically the class properties.

**Comment by marco-c** [comment]
Three outstanding problems:
- the linter fails with class properties, but we'd like to use them;
- Travis fails with Node.js v0.12;
- I don't know why the test change was needed.

**Comment by marco-c** [comment]
The test change is needed because babel 6 changed the way they export the default class.
There is a plugin to get the old behavior back, but I think we don't need it.

**Comment by marco-c** [comment]
@digitarald do we still want to support Node.js v0.12?

**Comment by digitarald** [comment]
I don't see a need, but @darkwing brought up before. If we depend on 4 we
need to document it in the contribution guidelines.

On Thursday, December 3, 2015, Marco <notifications@github.com> wrote:

> @digitarald <https://github.com/digitarald> do we still want to support
> Node.js v0.12?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/platatus/pull/209#issuecomment-161640959>.
>


-- 
Web Craftsman | harald@mozilla.com | @digitarald [skype/irc/line] |
+1-650-963-6612 [wapp/telegram]

**Pull request merged**